packages/react-instantsearch/rollup.config.js

```diff
@@ -0,0 +1,112 @@
+/* eslint-disable import/no-extraneous-dependencies */
+import resolve from 'rollup-plugin-node-resolve';
+import commonjs from 'rollup-plugin-commonjs';
+import babel from 'rollup-plugin-babel';
+import globals from 'rollup-plugin-node-globals';
+import replace from 'rollup-plugin-replace';
+import uglify from 'rollup-plugin-uglify';
+import filesize from 'rollup-plugin-filesize';
+/* eslint-enable import/no-extraneous-dependencies */
+
+const clear = x => x.filter(Boolean);
+
+const version = process.env.VERSION || 'UNRELEASED';
+const algolia = '© Algolia, inc.';
+const link = 'https://community.algolia.com/react-instantsearch';
+const license = `/*! ReactInstantSearch ${version} | ${algolia} | ${link} */`;
+
+const plugins = [
+  babel({
+    exclude: 'node_modules/**',
+    // see: https://github.com/rollup/rollup-plugin-babel#helpers
+    plugins: ['external-helpers'],
+  }),
+  resolve({
+    browser: true,
+    preferBuiltins: false,
+  }),
+  commonjs({
+    // Rollup only support ES modules so we need to transform commonjs modules to ES one.
+    // Can be magically done sometimes, but Rollup needs a bit of help in some use cases.
+    // see doc: https://github.com/rollup/rollup-plugin-commonjs#custom-named-exports
+    // see API: https://github.com/algolia/algoliasearch-helper-js/blob/develop/index.js
+    namedExports: {
+      'algoliasearch-helper': [
+        'version',
+        'AlgoliaSearchHelper',
+        'SearchParameters',
+        'SearchResults',
+        'url',
+      ],
+    },
+  }),
+  // Useful for define process.env variable, inside the AlgoliaSearchClient we
+  // have a dependency on it for RESET_APP_DATA_TIMER.
+  // see: http://bit.ly/2zGLrsu (Permalink was too long)
```

Comment:
Would still put `https://github.com/algolia/algoliasearch-client-javascript/blob/bb5d2ce16cf4b5a6e6de8ebae2de2880f783d967/src/AlgoliaSearchCore.js#L13`. eslint is fine with long comments of urls.

Reply:
Also that process.env in the client should already be resolved/set in the build of the js client no?

Reply:
No really because Rollup will resolve the file [algoliasearchLite](https://github.com/algolia/algoliasearch-client-javascript/blob/cd22e01862f2612b22b595b1815968ad1daec148/src/browser/builds/algoliasearchLite.js) according to the [import](https://github.com/algolia/react-instantsearch/blob/master/packages/react-instantsearch/dom.js#L3) & the [`package.json`](https://github.com/algolia/algoliasearch-client-javascript/blob/master/package.json#L9). This entry point require `AlgoliaSearchCore` and so the process.env is still present when Rollup do the build.

What we could do also if you are sure to not use Node globals elsewhere, is to replace **all** `process.env` calls with their corresponding values at build time (like we do with `process.env.NODE_ENV`). We can gain maybe 1-2k of payload size without using the Node globals plugin.

Reply:
I'll replace that in the client by an option instead.

Reply:
Ok so we just ensure process.env is defined and then we use the default value in the client. WFM.